File: 66.加一.py
#
# @lc app=leetcode.cn id=66 lang=python3
#
# [66] 加一
#

# @lc code=start
from typing import List
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i=len(digits)-1
        digits[-1]+=1
        while(digits[i]>9):
            digits[i]%=10
            i-=1
            if i<0:return [1]+digits
            digits[i]+=1
        return digits
# @lc code=end

File: 66.加一.cpp
/*
 * @lc app=leetcode.cn id=66 lang=cpp
 *
 * [66] 加一
 */

// @lc code=start
#include<vector>
using namespace std;
class Solution {
public:
    void func(vector<int>& digits, int start) {
        if(start == -1) {
            digits.insert(digits.begin(), 1);
            return;
        }
        ++digits[start];
        if(digits[start] / 10){
            digits[start] %= 10;
            func(digits, start - 1);
        }
    }
    vector<int> plusOne(vector<int>& digits) {
        func(digits, digits.size() - 1);
        return digits;
    }
};
// @lc code=end

